### PR TITLE
Image support

### PR DESCRIPTION
**Pull request by ngscheurich**
This adds support for embedding images in workbooks. It technically does
two main things:

- Implements the changes introduced by [proactively's image support
  branch](https://github.com/proactively/elixlsx/tree/image-support-no-reformat) on top of the most recent commit to master
- Adds `width` and `height` options to `Elixlsx.Image` so that it's
  possible to configure how many columns/rows an image should occupy

All the credit for this goes to [proactively](https://github.com/proactively).

---

_Note:_ This PR duplicates most of the work in https://github.com/xou/elixlsx/pull/79. I needed a working copy of this ASAP for a project, so I created this branch. Feel free to close this in favor of #79 if work is still being done there!

### CODE REVIEW

File: lib/elixlsx/xml_templates.ex [hunk 1]
```diff
@@ -432,6 +470,99 @@ defmodule Elixlsx.XMLTemplates do
...
+                        r:embed="rId#{drawing_id}" cstate="print">
+                </a:blip>
+                <a:stretch>
+                    <a:fillRect/>
+                </a:stretch>
+            </xdr:blipFill>
+            <xdr:spPr>
+                <a:xfrm>
+                    <a:off x="1812547" y="3503925"/>
+                    <a:ext cx="240431" cy="237600"/>
```
Comment (xou) [inline lib/elixlsx/xml_templates.ex:526]: These numbers seem oddly specific. Can you add a comment where they come from?

File: example.exs [hunk 2]
```diff
@@ -136,7 +136,20 @@ sheet6 =
...
 
+# Images
+sheet7 = %Sheet{
+  name: "Images",
+  rows: List.duplicate(["A", "B", "C", "D", "E"], 5)
+}
+
+sheet7 =
+  sheet7
+  |> Sheet.insert_image(0, 5, "ladybug-3475779_640.jpg")
```
Comment (xou) [inline example.exs:147]: Can you verify that this is a picture we can actually use in an open source project?
If required, can you add an attribution notice (not a legal expert, but maybe just a ladybug-3475779_640.jpg.NOTICE txt file or something)

File: lib/elixlsx/sheet.ex [hunk 3]
```diff
@@ -110,6 +114,8 @@ defmodule Elixlsx.Sheet do
   """
   def set_at(sheet, rowidx, colidx, content, opts \\ [])
       when is_number(rowidx) and is_number(colidx) do
+    sheet = maybe_extend(sheet, rowidx, colidx)
+
     cond do
       length(sheet.rows) <= rowidx ->
         # append new rows, call self again with new sheet
```
Comment (xou) [inline lib/elixlsx/sheet.ex:121]: The below code is now mostly handled by maybe_extend / dead, if I'm not mistaken?

**Comment by proactively** [comment]
Thank you so much for picking this up. The photo is a Free for commercial use
No attribution required photo: https://pixabay.com/photos/ladybug-insect-beetle-nature-3475779/

**Comment by ngscheurich** [comment]
@proactively My pleasure. My company needed this functionality ASAP for a client app, so it was great to find that most of the work had been done! I'll try and take a look at this and get it shaped up in the near future.

**Comment by grrrisu** [comment]
@ngscheurich @xou Any news on this PR? Would be an awesome feature.

**Comment by frerich** [comment]
For what it's worth, I'd love to have a release with this feature :-)

**Comment by sergey-chechaev** [comment]
Do you have any updates on this PR?

**Comment by ngscheurich** [comment]
Closing this in favor of #124. Thanks for your work, [@ustrajunior](https://github.com/ustrajunior)!

**Pull request closed without merging**